File: project/code/parseEnron.hs

import System.Directory
import Data.List.Split
import Data.String.Utils
import Data.List
import Text.Regex
import Data.Maybe

s = 
  [
    "^(to :|cc :|subject :|Subject:)", 
    "|[0-9]{2} : [0-9]{2} (am|pm)",
    "|- - - -"
  ]

ts = concat s

formatString chunk = 
  let 
    lines             = splitOn "\n" chunk 
    doesNotMatchRegex = (not . isJust . (matchRegex $ mkRegex ts))
  in
    intercalate "\n" $ filter doesNotMatchRegex lines

main :: IO ()
main = do 
  all <- getDirectoryContents "samples"
  print all
  {-text <- readFile "samples/file5.txt"-}
  {-writeFile "samples/newfile5.txt" $ formatString text-}
  {-putStrLn $ show $ formatString text-}

File: project/code/parseEnron.js


var fs = require('fs');

var re = 
    "^(to :|cc :|subject :|Subject:)" +
    "|[0-9]{2} : [0-9]{2} (am|pm)" +
    "|- - - -";


var options = { encoding: 'utf-8' };

fs.readFile('samples/file1.txt', options, function (err, data) {
  if (err) console.log(err);
  else {
    var processed = processString(data);
    fs.writeFile('samples/newfile1.txt', processed, options, 
      function (err, data) {
        if (err) console.log(err);
        else console.log('File saved!');
      }
    );
  }
});

var processString = function (str) {
  var lines = str.split("\n");
  var filtered = lines.filter(function (line) {
    return !line.match(re);
  });
  return filtered.join("\n");
};

